test_nnf.py

```diff
@@ -754,3 +754,24 @@ def test_dsharp_compile_converting_names(sentence: And[Or[Var]]):
         assert all(isinstance(name, str) for name in compiled.vars())
         if sentence.satisfiable():
             assert sentence.equivalent(compiled)
+
+
+@given(NNF())
+def test_tseitin(sentence: nnf.NNF):
+    sentence = sentence.simplify()
+    assume(len(sentence.vars()) <= 6)
+    assume(len(sentence.vars()) >= 2)
+
+    T = sentence.simplify()
+    T = tseitin.to_cnf(T)
+
+    # TODO: Once forgetting/projection is implemented, do this more complete check
+    # aux = filter(lambda x: 'aux' in str(x.name), T.vars())
+    # assert T.forget(aux).equivalent(sentence)
```

Comment:
I think it comes down to assumptions on equivalence -- my initial thought was that they need to have the same variables, as equivalence can be cast in terms of having the exact same models. Looking at the implementation, I can see that you've taken the measure of evaluating equivalence over the set of vars corresponding to the union of the two.

Maybe `projected_equivalence` as a new function that forgets all vars not in the intersection? Doesn't target the aux variables specifically, but could get the job done.

If aux vars have a special treatment as their own class, then it makes sense to forget them with a single simple call (so skipping the collection of aux vars here). Then it could be just a matter of `assert T.forget_aux().equivalent(sentence)`.

Reply:
I think it should be straight forward to make this swap once we have forgetting in place. Just taking the intersection is dangerous (related to #11 ), but with aux vars explicitly treated we should be set (cf. #7 ).

Resolving for now (to be resolved w/ #7).